design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,188 @@
+
+This design proposal outlines the deployment process for the ClusterLink project on a Kubernetes cluster by the site administrator. It relies on a dedicated K8s operator to make the deployment more user-friendly, simplified, and native to K8s users.
+
+## Background
+
+### Motivation and problem space
+
+This design proposal aims to simplify and enhance the deployment of the ClusterLink project.
+By leveraging K8s native mechanisms, such as operators and CRDs, the objective is to establish a user-friendly deployment process.
+Currently, ClusterLink deployment involves using the cl-adm CLI, which generates a large YAML file that contains the control plane and dataplane deployments, services, and RBAC configurations.
+In the new design, this process will be managed by the ClusterLink operator, which has the following advantages:
+
+* It consumes a simpler and more concise YAML configuration.
+* The use of CRD and the operator aids in automating deployment changes over time.
+* The proposal differentiates between fabric and site administrators and provides a more secure deployment model via k8s RBAC.
+
+### Impact and desired outcome
+
+Implementing this proposal is expected to improve and simplify the deployment process for the ClusterLink users.
+Aligning the deployment process with K8s standards will particularly benefit use cases involving central control management.
+
+### Prior discussion and links
+
+Not applicable.
+
+## User/User Story
+
+The below user stories describe sample needs of a site administrator. Fabric administrators' requirements aren't elaborated.
+
+1. As a site administrator, I want to join my cluster with an existing fabric. After receiving the credentials (e.g., fabric certificate and site keypair, I would like to provide minimal additional configuration needed to deploy ClusterLink to the site/cluster I manage.
+1. As a site administrator, I would like to scale my ClusterLink deployment to meet growing demand (e.g., bandwidth) for cross-site communication.
+1. As a site administrator, I would like to completely remove ClusterLink from my cluster, restoring its previous state.
+
+## Goals
+
+This design document should:
+
+* Define the steps for deploying a ClusterLink gateway to a K8s cluster.
+* Define the task functions of the ClusterLink deployment operator for setting up the ClusterLink components in the cluster.
+* Define the CRD (Custom Resource Definition) for the ClusterLink deployment operator.
+
+## Non-Goals
+
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+* This document focuses only on the deployment of ClusterLink to the K8s cluster.
+The deployment to another environment, such as VMs, is out of scope.
```

Comment:
```suggestion
 The deployment to another environment, such as VMs, is out of scope.
```
Should be combined with previous line or added in a separate line with indentation.